src/functions/authoriser.ts

```diff
@@ -20,65 +19,64 @@ const authoriser: any = async (event: any, context: Context): Promise<Policy | u
 
   if (!event.authorizationToken) {
     context.fail(`No authorization methods provided.\nEvent dump:\n${event}\nContext dump:\n${context}`);
-    return undefined;
+    return;
```

Comment:
Since these no longer return undefined, should the return type now be `Promise<Policy | void>`?

Reply:
nice catch thanks,
I think the original return was there to stop the execution early.
It is probably not required as if we fail to get the jwt token we throw and return an unauthorised policy however context.fail does not stop the execution so we will probably need to throw an error here too. We catch StatusCodeError and AuthorizationError so we may want to throw that.
We can probably discuss about the preferred approach when you have some time please?